pkgs/games/factorio/utils.nix

```diff
@@ -5,7 +5,7 @@ with stdenv.lib;
 {
   mkModDirDrv = mods: # a list of mod derivations
     let
-      recursiveDeps = modDrv: [modDrv] ++ optionals (modDrv.deps == []) (map recursiveDeps modDrv.deps);
+      recursiveDeps = modDrv: [modDrv] ++ optionals (modDrv.deps != []) (map recursiveDeps modDrv.deps);
```

Comment:
Why is this written like this? `map` over an empty list should be empty so there ought to be no reason not to write it as:

```nix
recursiveDeps = modDrv: [modDrv] ++ map recursiveDeps modDrv.deps
```

Am I missing something here?

Reply:
I don't see any reason why the `optionals` needs to be there. Thanks for the review!